src/main/resources/org/jenkinsci/plugins/workflow/flow/GlobalDefaultFlowDurabilityLevel/global.jelly

```diff
@@ -0,0 +1,17 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+    <f:section title="Pipeline Speed/Durability Settings">
```

Comment:
I strongly recommend _against_ using "Speed" in this configuration. I believe including "Speed" will encourage otherwise uneducated administrators and users to make a durability sacrifice, thinking their Pipelines will somehow run magically faster, with the "speedier" setting

If this section is simply "Pipeline Durability Settings" then I think we (CloudBees and Jenkins docs folk) can do the appropriate education in knowledgebase articles for fine-tuning Jenkins master performance to those who will actually benefit from the durability/speed trade-off..

Reply:
@rtyler That's actually what they *should* be doing and what we *want* them to do.  As long as you aren't randomly SIGKILL-ing Jenkins, you should be using PERFORMANCE_OPTIMIZED mode except for critical production deploys.  And there's nothing "magical" about things going exponentially faster when you're not writing a ton of redundant data to disk. 😉

Reply:
Basically all Jenkins instances are critical production deploys in my opinion. Jenkins is the life-blood of many organizations's delivery pipeline. 

My concern with this phrasing is that if the administrator is not well versed in the implications of these settings (read: everybody), conveying that it will change "Pipeline speed" is very likely going to be understood from the sense of Pipeline execution speed (wall-clock time): "Oh all my Pipelines will finish sooner now, this is like pushing in the Turbo button on my PC, great!" Sure that may occur, but the thrust of this setting's point as it should be conveyed to the user, is that this will change the Durability contract.
 
The vast majority of a Pipeline's wall-clock time in ci.jenkins.io is running Maven, enticing derpy admins like me to change this Durability setting for improved "Pipeline speed" is misleading.

Reply:
@rtyler Ah!  I think my phrasing was unclear -- what I mean by "production deploy" is a pipeline that is deploying code to a production system -- I consider *all* Jenkins instances to be production servers themselves and prioritize stability accordingly.   Downstream PRs add the ability to customize the durability mode on a per-pipeline or per-branch basis.   This means you can set up normal PR builds to just run as fast and light as possible to ensure that you don't overload a master (improving stability), but ensure that the most critical jobs *which deploy apps to prod* get the utmost level of protection.

The downstreams also explicitly test for the worst-case scenario to ensure that we gracefully handle it -- that scenario is an **unplanned** outage of a master while a pipeline is running in the performance_optimized mode.  In that scenario, the worst-case behavior is that the pipeline behaves like a freestyle build -- fails cleanly, retaining the build record and logs.

Only the transitory pipeline data is impacted by durability settings (persistence of the flowgraph, run itself, and program.dat) -- and we have mechanisms (with testing) to ensure they gracefully fail when data isn't persisted. 

Basically, I'm signing off that the chain of PRs should at no point reduce the stability of masters and should improve it overall, because it will prevent a class of outage caused by less-than-optimal pipeline code demanding excessive IO.

Is there a better way to communicate all the above?  I can understand why it's confusing.

> Oh all my Pipelines will finish sooner now, this is like pushing in the Turbo button on my PC, great

That's pretty much the behavior I expect for a lot of enterprise masters, and have a short list that will see that outcome.  

The difference between those and the ci.jenkins.io masters is that we have some of the world's greatest experts in Jenkins and Pipeline tuning our pipelines.  This allows us to easily minimize the overheads so that most of our time actually *is* running sh/bat steps -- but most organizations don't have easy access to the expertise in Jenkins that the Jenkins project does (naturally enough).  The results unfortunately speak for themselves. 

Basically: my goal is to make pipelines easier for people by removing some nasty performance surprises that hang people up -- and generally make them much lighter on the masters (reducing performance-related outages).